crates/linker/src/pe.rs

```diff
@@ -724,6 +1030,54 @@ fn write_section_header(
     data[section_header_start..][..header_array.len()].copy_from_slice(&header_array);
 }
 
+fn increase_number_of_sections_help(
+    input_data: &[u8],
+    new_sections: &[[u8; 8]],
+    output_file: &Path,
+) -> MmapMut {
+    use object::read::pe::ImageNtHeaders;
+
+    let dos_header = object::pe::ImageDosHeader::parse(input_data).unwrap();
+    let mut offset = dos_header.nt_headers_offset().into();
+    let header_offset = offset;
+    let number_of_sections_offset = header_offset + 4 + 2;
```

Comment:
Where do these numbers come from? The PE file format?

Reply:
yes. it's an offset into the header struct. Sadly we can't nicely get that offset in rust today, the macro proposed here  https://internals.rust-lang.org/t/get-the-offset-of-a-field-from-the-base-of-a-struct/14163/6 does not seem super solid.

Reply:
but I was able to refactor this away